File: csharp/rna-transcription/RnaTranscription.cs
using System;
using System.Linq;

public static class RnaTranscription
{
    public static string ToRna(string nucleotide)
    {
        if (nucleotide.Equals(""))
        {
            return "";
        }
        else
        {
            return string.Concat(nucleotide.Select(nuc => transcribeNucleotide(nuc)));
        }
    }

    private static char transcribeNucleotide(char nucleotide)
    {
        switch (nucleotide)
        {
            case 'C':
                return 'G';
            case 'G':
                return 'C';
            case 'T':
                return 'A';
            case 'A':
                return 'U';
            default:
                return '?';
        }
    }
}

File: fsharp/rna-transcription/RnaTranscription.fs
module RnaTranscription


let rnaTranscriptor dnaNucleotide = 
    match dnaNucleotide with
    | 'G' -> "C"
    | 'C' -> "G"
    | 'T' -> "A"
    | 'A' -> "U"
    | _ -> ""

let toRna (dna: string) = 
    dna |> Seq.map rnaTranscriptor |> String.concat ""